tests/src/JIT/HardwareIntrinsics/X86/General/VectorArgs.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+// This test case is ported from S.N.Vector counterpart 
+// https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/SIMD/VectorArgs.cs
+
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+internal partial class IntelHardwareIntrinsicTest
+{
+    private const int Pass = 100;
+    private const int Fail = -1;
+
+    internal class VectorArg128
+    {
+        private Vector128<float> _rgb;
+
+        public VectorArg128(float r, float g, float b)
+        {
+            float[] temp = new float[4];
+            temp[0] = r; temp[1] = g; temp[2] = b;
+            unsafe
+            {
+                fixed (float* ptr = temp)
+                {
+                    _rgb = Sse.LoadVector128(ptr);
+                }
+            }
+        }
+
+        public VectorArg128(Vector128<float> _rgb)
+        { this._rgb = _rgb; }
+
+        public VectorArg128 Change(float f)
+        {
+            Vector128<float> t = Sse.SetAllVector128(f);
+            return new VectorArg128(Sse.Add(t, _rgb));
+        }
+
+        public Vector128<float> RGB { get { return _rgb; } }
+    }
+
+    internal class VectorArg256
+    {
+        private Vector256<float> _rgb;
+
+        public VectorArg256(float r, float g, float b)
+        {
+            float[] temp = new float[8];
+            temp[0] = r; temp[1] = g; temp[2] = b;
+            unsafe
+            {
+                fixed (float* ptr = temp)
+                {
+                    _rgb = Avx.LoadVector256(ptr);
+                }
+            }
+        }
+
+        public VectorArg256(Vector256<float> _rgb)
+        { this._rgb = _rgb; }
+
+        public VectorArg256 Change(float f)
+        {
+            Vector256<float> t = Avx.SetAllVector256(f);
+            return new VectorArg256(Avx.Add(t, _rgb));
+        }
+
+        public Vector256<float> RGB { get { return _rgb; } }
+    }
+
+    unsafe static int Main()
+    {
+        int returnVal = Pass;
+
+        if (Sse41.IsSupported)
```

Comment:
We should probably support this test on SSE as well. The code for extracting an element isn't too complicated (just a shuffle and convert to scalar).

Reply:
Thanks, will do